src/AzureAuth/CommandMain.cs

```diff
@@ -159,6 +162,40 @@ public CommandMain(CommandExecuteEventData eventData, ILogger<CommandMain> logge
         [Option(ConfigOption, "The path to a configuration file.", CommandOptionType.SingleValue)]
         public string ConfigFilePath { get; set; }
 
+        /// <summary>
+        /// Gets or sets the cache file name. Only available on Windows.
+        /// </summary>
+        [Option(CacheOption, "Override the default cache file location. This option is only available on Windows.", CommandOptionType.SingleValue, ShowInHelpText = IsWindows)]
+        [LegalFilePath]
+        public string CacheFilePath
+        {
+            get
+            {
+                // Check command parameter first.
+                if (!string.IsNullOrEmpty(this.cacheFilePath))
+                {
+                    return this.cacheFilePath;
+                }
+
+                // Check environment variable.
+                string envCacheFile = this.env.Get(EnvVars.AZUREAUTH_CACHE_FILE);
+                if (!string.IsNullOrEmpty(envCacheFile))
+                {
+                    return envCacheFile;
+                }
+
+                // Use default cache file path.
+                string appData = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
+                string absolutePath = this.fileSystem.Path.Combine(appData, ".IdentityService", $"msal_{this.authSettings.Tenant}.cache");
+                return absolutePath;

```

Comment:
Not blocking, but generally I'd prefer to see logic like this extracted into a method, and then used only to memoize (cache) the result using a private member to back this Property. It's a small performance gain to only access the Env once. We are going to access this getter at least twice - once to validate, and once to use. But the gain here is small enough I don't want to block on this now.